swap_meet/vendor.py

```diff
@@ -1,2 +1,117 @@
+        # Otherwise, remove this item from inventory and return this removed item
+        if item_remove not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item_remove)
+            return item_remove
+        
+    def get_by_id(self, id):
+        # Create an Item instance passing in an id value
+        test_item = Item(id)
+        # Loop through list of instances stored in attribute inventory
+        for item in self.inventory:
+            # If attribute id in an instance has same value as attribute id in Item instance
+            # Return this instance. Otherwise, return None
+            if item.id == test_item.id:
+                return item
+        return None
+
+            
+    def swap_items(self, other_vendor, my_item, their_item):
+        # If user's Item instance doesn't exist in own inventory or friend's Item instance doesn't exist
+        # in friend's inventory, return False
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        # Otherwise, remove user's Item instance and add it to friend's inventory
+        # Remove friend's Item instance and add it to user's inventory
+        # Return True
+        else:
+            self.inventory.remove(my_item)
+            other_vendor.inventory.append(my_item)
+            other_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+    
+    def swap_first_item(self, other_vendor):
+        # If user's own inventory or friend's inventory is empty, return False
+        if len(self.inventory) == 0 or len(other_vendor.inventory) == 0:
+            return False
+        else:
+            # Otherwise, grab first Item instance in user's and friend's inventories
+            user_first_item = self.inventory[0]
+            friend_first_item = other_vendor.inventory[0]
+            # Call on the swap_items method to swap these first Item instances between user and friend
+            self.swap_items(other_vendor, user_first_item, friend_first_item)
+            # self.inventory.remove(user_first_item)
+            # other_vendor.inventory.remove(friend_first_item)
+            # other_vendor.inventory.append(user_first_item)
+            # self.inventory.append(friend_first_item)
+            return True
+        
+    def get_by_category(self, category):
+        # Note: category of type string
+        # Create an empty list called "resulting_list"
+        resulting_list = []
+        # Loop through each Item instance in inventory
+        for item in self.inventory:
+            # If an instance's category (aka class name) same as provided category,
+            # add this instance to resulting_list
+            if item.get_category() == category:
+                resulting_list.append(item)
+        # Return updated resulting_list
+        return resulting_list
```

Comment:
Notice this pattern of: 
```python
result_list = []
for element in source_list:
    if some_condition(element):
        result_list.append(element)
```

can be rewritten using a list comprehension as:
```python
result_list = [element for element in source_list if some_condition(element)]
```

Which here would look like:
```python
resulting_list = [item for item in self.inventory if item.get_category() == category]
return resulting_list
```

At first, this may seem more difficult to read, but comprehensions are a very common python style, so I encourage y’all to try working with them!